### PR TITLE
Add module tests for config and ingest/publish

### LINKED ISSUE #1400: Module tests: configuration and option handling
Add unit and small integration tests that cover configuration parsing, validation, defaults, and invalid/edge-case combinations. Tests should:
- Validate default values and that documented behaviors are enforced
- Cover invalid combinations and error messages returned by parsing/validation
- Test missing/null/empty fields and how the code falls back to defaults
- Include any fixtures or builders for config objects in the test code (do not create a separate issue for tests)

This work depends on shared testing infra from the testing-infrastructure sub-issue.

Acceptance criteria:
- Comprehensive unit tests for config parsing/validation functions
- Tests included in the same PR as any necessary small code changes

### PR DESCRIPTION
**Pull request by henry0715-dev**
- The root certificate loader now validates empty inputs and returns an error if no valid certificates are added.
- Move settings initialization out of main to improve readability and reuse configuration loading logic.

Close #1400

### CODE REVIEW

File: src/comm/publish/tests.rs [hunk 1]
```diff
@@ -1019,3643 +2072,705 @@ async fn request_range_data_with_protocol() {
...
-    {
-        let (mut send_pub_req, mut recv_pub_resp) =
-            publish.conn.open_bi().await.expect("failed to open stream");
-        let ftp_store = db.ftp_store().unwrap();
-        let send_ftp_time = Utc::now().timestamp_nanos_opt().unwrap();
-        let ftp_data =
-            bincode::deserialize::<Ftp>(&insert_ftp_raw_event(&ftp_store, SENSOR, send_ftp_time))
-                .unwrap();
+#[tokio::test]
+async fn request_raw_events() {
```
[the remark this answers is not in the document]
Comment (henry0715-dev) [reply src/comm/publish/tests.rs]: I’ve added the requested test code.

**Comment by dayeon5470** [comment]
It would also be good to add tests for `check_for_duplicate_connections`in peer.rs and for `filter_ip` in publish/tests.rs.

**Comment by henry0715-dev** [comment]
I will split this large PR into smaller units of work, so this PR will be closed.
